### PR TITLE
Headers only ingest [VS-1187]

### PR DESCRIPTION
**Pull request by mcovarr**
Integration run [in progress](https://app.terra.bio/#workspaces/gvs-dev/GVS%20Integration/job_history/544fb86b-ffb7-447c-b380-fdefce10be99).

Does away with the `STARTED` and `FINISHED` sample load statuses to more explicitly record what work has actually been done for a sample: `REFERENCES_LOADED`, `VARIANTS_LOADED` or `HEADERS_LOADED`. Legacy `FINISHED` and `STARTED` states are recognized and handled appropriately (short circuiting data load and being ignored, respectively).

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/ingest/CreateVariantIngestFiles.java [hunk 1]
```diff
@@ -243,40 +255,48 @@ public void onTraversalStart() {
...
+            LoadStatus.LoadState state = loadStatus.getSampleLoadState(sampleId, loadHeadersOnly);
             if (state == LoadStatus.LoadState.COMPLETE) {
                 logger.info("Sample id " + sampleId + " was detected as already loaded, exiting successfully.");
                 System.exit(0);
+            } else if (loadHeadersOnly && state == LoadStatus.LoadState.HEADERS_WRITTEN) {
+                logger.info("Sample id " + sampleId + " has already had its headers written, exiting successfully.");
+                System.exit(0);
             } else if (state == LoadStatus.LoadState.PARTIAL) {
-                if (enableReferenceRanges) {
+                if (enableReferenceRanges && !loadHeadersOnly) {
```
Comment (koncheto-broad) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/ingest/CreateVariantIngestFiles.java:266]: This and the following two changes are fine, but it feels like this entire block can be simplified by just wrapping it all in a  `if (!loadHeadersOnly) { ... }` block to me.  I'm not sure what the value is in adding that logic to every individual condition in this stretch

**Review by koncheto-broad — APPROVED** [review]
LGTM.  Small suggestion that is optional and should not block merging

File: scripts/variantstore/wdl/GvsAssignIds.wdl [hunk 2]
```diff
@@ -16,6 +16,7 @@ workflow GvsAssignIds {
     Boolean samples_are_controls = false
 
     Boolean process_vcf_headers = false
+    Boolean load_headers_only = false
```
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsAssignIds.wdl:19]: One thought (as discussed) would be to replace these two booleans with one that loads only the headers and one that loads the rest of the data (ref_ranges and vets). So if we wanted to load *only* the headers we'd set the first boolean to true, and if we wanted to load *only* the rest of the data (ref_ranges and ves) we'd only set the second to true, and if we wanted to load everything (like for AoU) we'd set both booleans. Both set to false is an error condition. 
Just a thought.

**Review by gbggrant — APPROVED** [review]
LGTM

File: scripts/variantstore/wdl/GvsImportGenomes.wdl [hunk 3]
```diff
@@ -66,6 +68,14 @@ workflow GvsImportGenomes {
   String effective_gatk_docker = select_first([gatk_docker, GetToolVersions.gatk_docker])
   String effective_git_hash = select_first([git_hash, GetToolVersions.git_hash])
 
+  if (!load_vcf_headers && !load_vet_and_ref_ranges) {
+    call Utils.TerminateWorkflow as MustLoadAtLeastOneThing {
+      input:
+        message = "GvsBulkIngestGenomes called with both load_vcf_headers and load_vet_and_ref_ranges set to false",
```
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsImportGenomes.wdl:74]:
```suggestion
        message = "GvsImportGenomes called with both load_vcf_headers and load_vet_and_ref_ranges set to false",
```

Comment (mcovarr) [reply scripts/variantstore/wdl/GvsImportGenomes.wdl:74]: whoops busted copy/pasting

**Review by koncheto-broad — APPROVED** [review]
LGTM. I very much like having the states separated out for headers, variants, and refs

**Pull request merged**